bugbug/tools/code_review.py

```diff
@@ -680,6 +717,31 @@ def collect_function_definitions(function_declarations, target_element, definiti
         )
 
 
+def gather_line_context(line_context):
+    r"""Reformat the line_context list.
+
+    From [[line, file, function],...] to [[gathered_line, file, function],...]
+    to avoid repetitions of the same tuple (file, function) in the list.
+    'gathered_line' is a str that concatenates the 'line' (with separator '\n')
+    that required the same function.
+    """
+    file_dir = {}
+
+    for line, file, func in line_context:
+        if file not in file_dir:
+            file_dir[file] = {}
+        if func not in file_dir[file]:
+            file_dir[file][func] = []
+        file_dir[file][func].append(line)
+
+    gathered_context = []
+    for file in file_dir:
+        for func in file_dir[file]:
+            gathered_requested_lines = "\n".join(file_dir[file][func])
+            gathered_context += [[gathered_requested_lines, file, func]]
```

Comment:
Why this is a list, not a tuple?
```suggestion
            gathered_context += [(gathered_requested_lines, file, func)]
```